yarpl/include/yarpl/Refcounted.h

```diff
@@ -14,32 +14,24 @@ namespace yarpl {
 /// of boost::intrusive_ptr<>, except that we have virtual methods
 /// anyway, and want to avoid argument-dependent lookup.
 ///
-/// NOTE: only derive using "virtual public" inheritance.
+/// NOTE: Only derive using "virtual public" inheritance.
 class Refcounted {
  public:
-#if !defined(NDEBUG)
-  Refcounted();
-  virtual ~Refcounted();
-
-  // Return the number of live refcounted objects.  For testing.
-  static std::size_t objects();
+  virtual ~Refcounted() = default;
```

Comment:
Should we have a warning, logging (at least in the debug build), if we delete a Refcounted object whose reference count is greater than 1?